synapse/cortex.py

```diff
@@ -1206,6 +1222,120 @@ async def coreQueuePop(self, name, offs):
     async def coreQueueSize(self, name):
         return self.multiqueue.size(name)
 
+    @s_nexus.Pusher.onPushAuto('tag:model:set')
+    async def setTagModel(self, tagname, name, valu):
+        '''
+        Set a model specification property for a tag.
+
+        Arguments:
+            tagname (str): The name of the tag.
+            name (str): The name of the property.
+            valu (object): The value of the property.
+        '''
+        meta = self.taghive.get(tagname)
+        if meta is None:
+            meta = {}
+
+        meta[name] = valu
+        reqValidTagModel(meta)
```

Comment:
Do we want this to be on the top half of the nexus func?

Reply:
This *must* be in the bottom half to ensure only valid changes make it in.  We could have it in both, but that would incur the extra lookup to get the meta dict in both places and would only really save us one nexus log entry per invalid call ( a definite edge case ).  I decided it wasn't worth it.